include/engine/engine_config.hpp

```diff
@@ -60,11 +64,17 @@ struct EngineConfig final
 {
     bool IsValid() const;
 
+    constexpr static const int UNLIMITED = -1;
+
     storage::StorageConfig storage_config;
-    int max_locations_trip = -1;
-    int max_locations_viaroute = -1;
-    int max_locations_distance_table = -1;
-    int max_locations_map_matching = -1;
+    int max_locations_trip = UNLIMITED;
+    int max_locations_viaroute = UNLIMITED;
+    int max_locations_distance_table = UNLIMITED;
+    int max_locations_map_matching = UNLIMITED;
+    int max_results_nearest = UNLIMITED;
+
+    double max_radius_when_bearings = UNLIMITED; // in meters
```

Comment:
This is due to our `static_r_tree` functions being overloaded for `double` (radius) and `int` max_entries... Error-prone yeah...